### ISSUE TITLE
Backslash in Prometheus label value does not work with dashboard variable

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [grafana/grafana#24478]
**What happened**:
A prometheus label **value** containing a backlash `\` causes an error when using the value via grafana dashboard variable. Example: `logfile="C:\test\logfile.txt"`

**What you expected to happen**:
There should be some way to make it work?

**How to reproduce it (as minimally and precisely as possible)**:
- Have a prometheus metric, with some label with a value containing a backlash, e.g. `backslashtest{logfile="C:\test\logfile.txt"}` , easiest created by textfile collector.
- Create a dashboard variable `Name`=`logfile`, `Type`=`Query`, `Datasource`=Prometheus, `Query`=`label_values(backslashtest, logfile)`, this gives:
![grafik](https://user-images.githubusercontent.com/33002073/81480962-334ba300-922d-11ea-8ccb-672bc5550adf.png)
- Try to query the metric using this variable, I got no working solution.

```
backslashtest{logfile="$logfile"}
```
or
```
backslashtest{logfile=~"$logfile"}
```

yields:

```
"1:23: parse error: unknown escape sequence U+006C 'l'"
```
![grafik](https://user-images.githubusercontent.com/33002073/81501017-351d7100-92d6-11ea-9799-2642f678f5ea.png)

**I tried all the [advanced-variable-format-options](https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options)**, neither worked.

```
backslashtest{logfile="${logfile:regex}"}
```

yields `"1:23: parse error: unknown escape sequence U+002E '.'"`

The only one that does not throw an error is `backslashtest{logfile="${logfile:percentencode}"}` but Prometheus returns no data for this.

Using the regex Operator `=~` makes no difference:

```
backslashtest{logfile=~"${logfile:regex}"}
backslashtest{logfile=~"${logfile:raw}"}
[...]
```

Only this works: `backslashtest{logfile="C:\\test\\logfile.txt"}`

Is there a way to use the variable?

**Anything else we need to know?**:

Query-Inspector:
```
{
  "response": {
    "status": "error",
    "errorType": "bad_data",
    "error": "1:24: parse error: unknown escape sequence U+006C 'l'",
    "message": "1:24: parse error: unknown escape sequence U+006C 'l'"
  }
}
```

- This issues for example arises because Windows filepaths use backlash as seperator, the metric source in my case is [`grop_exporter`](https://github.com/fstab/grok_exporter/)
- The raw exporter text actually escapes the backlash `backslashtest{logfile="C:\\test\\logfile.txt"}`
- When querying in prometheus graph, it needs to be escaped as well

Related?: #10555 #20869

**Environment**:
- Grafana version: `v6.7.3 and v7.0.0-beta3 (2169fc1a3f)`
- Data source type & version: Prometheus `2.18.1`
- OS Grafana is installed on: `Windows 10 1909`
- User OS & Browser: `Windows 10 1909` , Firefox 75

**Comment by Mario-Hofstaetter** [comment]
# Workaround

If anyone lands here via google, I was able to workaround this PAIN by substituting `\` with `/` directly in `grok_exporter` using the [`gsub` function](https://github.com/fstab/grok_exporter/blob/master/CONFIG.md#label-template-functions) like so:

```yaml
  labels:
    logfile: '{{gsub .logfile "\\\\" "/"}}'
```
so `backslashtest{logfile="C:/test/logfile.txt"}` works just fine in grafana 🎉 

I dunno if prometheus relabeling is able to do this, the docs there did not look that promising on first sight.

**Comment by torkelo** [comment]
backslashtest{logfile=~"${logfile}"}

Should work as it will use the prometheus specific regex escaping logic.

**Comment by Mario-Hofstaetter** [comment]
Are you able to reproduce?
Slightly extended original post, it is happening with `v6.7.3` and `v7.0.0-beta3`

**Comment by aocenas** [comment]
Seems like this may be a duplicate of https://github.com/grafana/grafana/issues/20869

**Comment by ivanahuckova** [comment]
Closed by  #26205
